node/container.go

```diff
@@ -123,29 +124,38 @@ func (a *HostAgent) StartContainer(cancel <-chan interface{}, serviceID string,
 	// Establish a connection to the master
 	masterClient, err := master.NewClient(a.master)
 	if err != nil {
-		logger.WithError(err).Error("Failed to get the system user account")
+		logger.WithField("master", a.master).WithError(err).Debug("Could not connect to the master")
 		return nil, nil, err
 	}
-	logger.WithField("systemUser", systemUser.Name).Debug("Got system user")
+	defer masterClient.Close()
 
-	svc, err := a.getService(serviceID)
+	evaluatedService, tenantID, err := masterClient.GetEvaluatedService(serviceID, instanceID)
 	if err != nil {
-		logger.WithError(err).Debug("Unable to get service")
+		logger.WithError(err).Error("Failed to get service")
 		return nil, nil, err
 	}
 
 	// pull the service image
-	imageUUID, imageName, err := a.pullImage(logger, cancel, svc.ImageID)
+	imageUUID, imageName, err := a.pullImage(logger, cancel, evaluatedService.ImageID)
 	if err != nil {
 		logger.WithError(err).Debug("Could not pull the service image")
 		return nil, nil, err
 	}
-	svc.ImageID = imageName
+	// Update the service with the complete image name
+	evaluatedService.ImageID = imageName
+
+	// get the system user
+	systemUser, err := masterClient.GetSystemUser()
+	if err != nil {
+		logger.WithError(err).Error("Failed to get the system user account")
+		return nil, nil, err
+	}
+	logger.WithField("systemUser", systemUser.Name).Debug("Got system user")
 
-	// set up the container
-	ctr, state, err := a.setupContainer(masterClient, svc, instanceID, imageUUID, imageName)
+	// get the container configs
+	ctr, state, err := a.setupContainer(tenantID, evaluatedService, instanceID, systemUser, imageUUID)
 	if err != nil {
-		logger.WithError(err).Debug("Could not setup container")
+		logger.WithError(err).Debug("Unable to get service")
```

Comment:
When I changed it, we were getting the service, not calling setup container. I think the mismatch is a leftover from merging with Glenn's code last night. I've put it back to the original, so it once again matches the surrounding code.